metsim/methods/mtclim.py

```diff
@@ -4,182 +4,103 @@
-        for h in np.arange(-hss, hss, dh):
-            cosh = np.cos(h)
-            sinh = np.sin(h)
-            cza = cosegeom * cosh + sinegeom
-            cbsa = sinh * bsg1 + cosh * bsg2 + bsg3
-            if (cza > 0.):
-                dir_flat_topa = dir_beam_topa * cza
-                am = 1.0 / (cza + 0.0000001)
-                if (am > 2.9):
-                    ami = min(max(int((np.cos(cza) / consts['RADPERDEG'])) - 69,0),20)
-                    am = consts['OPTAM'][ami]
-                sum_trans += np.power(trans, am) * dir_flat_topa
-                sum_flat_potrad += dir_flat_topa
-                # FIXME: This is a long conditional
-                if ((h < 0. and cza > coszeh and cbsa > 0.) or
-                        (h >= 0. and cza > coszwh and cbsa > 0.)):
-                    sum_slope_potrad += dir_beam_topa * cbsa
-            else:
-                dir_flat_topa = -1
-            tinystep = np.clip(((12 * 3600 + h * consts['SECPERRAD'])/dt), 0, tiny_step_per_day - 1)
-            tiny_rad_fract[i, tinystep] = max(dir_flat_topa, 0)
-        if daylength[i] and sum_flat_potrad > 0:
-            tiny_rad_fract[i] /= sum_flat_potrad
-        if daylength[i]:
-            tt_max0[i] = sum_trans / sum_flat_potrad
-            flat_potrad[i] = sum_flat_potrad / daylength[i]
-            slope_potrad[i] = sum_slope_potrad / daylength[i]
-        else:
-            tt_max0[i] = 0.
-            flat_potrad[i] = 0.
-            slope_potrad[i] = 0.
-    tt_max0[365] = tt_max0[364]
-    flat_potrad[365] = flat_potrad[364]
-    slope_potrad[365] = slope_potrad[364]
-    daylength[365] = daylength[364]
-    tiny_rad_fract[365] = tiny_rad_fract[364]
-    return tt_max0, flat_potrad, slope_potrad, daylength, tiny_rad_fract
-
-
-def calc_shortwave(df):
-    pass
+        df['swe'] = _simple_snowpack(df['precip'], df['t_min'], snowpack=loop_swe)
 
 
-def calc_srad_hum_it(df, tol=0.01, win_type='boxcar'):
+def calc_srad_hum(df: pd.DataFrame, sg: dict, tol=0.01, win_type='boxcar'):
     """
-    TODO
+    Calculate shortwave, humidity  
     """
-    window = np.zeros(n_days + 90)
-    t_fmax = np.zeros(n_days)
-    df['s_tfmax'] = 0.0
-
+    def _calc_tfmax(precip, dtr, sm_dtr):
+        b = params['B0'] + params['B1'] * np.exp(-params['B2'] * sm_dtr)
+        t_fmax = 1.0 - 0.9 * np.exp(-b * np.power(dtr, params['C']))
+        inds = np.nonzero(precip > options['SW_PREC_THRESH'])[0]
+        t_fmax[inds] *= params['RAIN_SCALAR']
+        return t_fmax 
+
+    window = np.zeros(params['n_days'] + 90)
     df['t_max'] = np.maximum(df['t_max'], df['t_min'])
     dtr = df['t_max'] - df['t_min']
     sm_dtr = pd.rolling_window(dtr, window=30, freq='D', 
                                win_type=win_type).fillna(method='bfill')
-    if n_days <= 30:
+    if params['n_days'] <= 30:
         print('Timeseries is shorter than rolling mean window, filling ')
         print('missing values with unsmoothed data')
```

Comment:
use the warnings module here instead of print statements. That way we can filter them  when running this code over large domains...